src/bos/common/clients/cfs/base.py

```diff
@@ -0,0 +1,69 @@
+#
+# MIT License
+#
+# (C) Copyright 2021-2025 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+from abc import ABC
+import logging
+
+from bos.common.clients.endpoints import BaseEndpoint
+from bos.common.utils import PROTOCOL
+
+LOGGER = logging.getLogger(__name__)
+
+SERVICE_NAME = 'cray-cfs-api'
+BASE_CFS_ENDPOINT = f"{PROTOCOL}://{SERVICE_NAME}/v3"
+
+
+class BaseCfsEndpoint(BaseEndpoint, ABC):
+    """
+    This base class provides generic access to the CFS API.
+    """
+    BASE_ENDPOINT = BASE_CFS_ENDPOINT
+
+    def get_items(self, **kwargs):
+        """Get information for all CFS items"""
+        return self.get(params=kwargs)
+
+    def update_items(self, data):
+        """Update information for multiple CFS items"""
+        return self.patch(json=data)
+
+
+class BasePagedCfsEndpoint(BaseCfsEndpoint, ABC):
+    """
+    This base class provides generic access to the CFS API, for endpoints that support paging.
+    """
+    ITEM_FIELD_NAME = ''
```

Comment:
Your abstract base class doesn't enforce ITEM_FIELD_NAME be filled in by anything that subclasses it, and I assume that's why you made it an abstract base class.

Doing something like this would cause an error to be thrown if  ITEM_FIELD_NAME  was undefined in a subclass.
    @property
    @abstractmethod
    def ITEM_FIELD_NAME(self) -> str:
        """
        This property must be overridden in subclasses to specify the item field name.
        """
        pass

Reply:
That is a better way to do it, I agree. I was copying how the BOS client did something similar, but I like your way better.